### ISSUE TITLE
create_item day=defined periods

### ISSUE THREAD

**Issue by sonicfray** [derek-damron/budgetr#37]
ability to specify like 15th of month, last day of month, last working day of month, or some others that would be paycheck-like frequencies.

**Comment by derek-damron** [comment]
Good call, I had been wrestling with how to keep the API clean while allowing users to specify specific days of the month.

What do you think about the following?

```
# 15th day of the month 
create_item(...,
            day = X,
            recurring = "monthly")
```

Where `X` is one of following:

```
day = n # The nth day of the month
day = "last" # The last day of the month
day = "last working" # The last working day of the month
```

**Comment by sonicfray** [comment]
That works for me.  Most companies will pay on 15th or the earliest business day before that and same for the last business day of the month.

**Comment by sonicfray** [comment]
Also another tweak would be having a recurring but one that only lasts between a start and end date.  It could be like a contract job or an expense like a car that has an end date.
